setup.py

```diff
@@ -44,16 +44,32 @@ def _minimal_ext_cmd(cmd):
             GIT_REVISION = out.decode('utf-8').strip()
         except OSError:
             GIT_REVISION = "Unknown"
+        try:
+            cmd = ['git', 'config', '--get', 'remote.origin.url']
+            out = _minimal_ext_cmd(cmd).decode('utf-8').strip()
+            start_idx = 0
+            for prefix in [b'git@github.com:', b'git://github.com/',
+                           b'http://github.com/', b'https://github.com/',
+                           b'ssh://git@github.com/']:
+                if out.startswith(prefix):
```

Comment:
I'd put in the code we both expect to work, but add a `print((type(out), type(prefix), GIT_REVISION)` in a commit, and see what you get (after rebase to get rid of 2.7 errors). I wonder if there is some weirdness with the version checkouts. But that information should help us debug it.